GANDLF/entrypoints/run.py

```diff
@@ -0,0 +1,291 @@
+        reset_flag, resume_flag = False, False
+
+    if reset_flag and resume_flag:
+        logging.warning(
+            "'reset' and 'resume' are mutually exclusive; 'resume' will be used."
+        )
+        reset_flag = False
+
+    # TODO: check that output_path is not passed in training mode;
+    #  maybe user misconfigured the command
+
+    logging.debug(f"{config=}")
+    logging.debug(f"{input_data=}")
+    logging.debug(f"{train_flag=}")
+    logging.debug(f"{model_dir=}")
+    logging.debug(f"{device=}")
+    logging.debug(f"{reset_flag=}")
+    logging.debug(f"{resume_flag=}")
+    logging.debug(f"{output_path=}")
+
+    try:
+        main_run(
+            data_csv=input_data,
+            config_file=config,
+            model_dir=model_dir,
+            train_mode=train_flag,
+            device=device,
+            resume=resume_flag,
+            reset=reset_flag,
+            output_dir=output_path,
+        )
+    except Exception:
+        # TODO: why so? Why not just default way when exception is printed
+        #  and process exits with code 1 automatically?
+        sys.exit("ERROR: " + traceback.format_exc())
+    print("Finished.")
+
+
+@click.command()
+@click.option(
+    "--config",
+    "-c",
+    required=True,
+    help="The configuration file (contains all the information related to the training/inference session)",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False),
+)
+@click.option(
+    "--input-data",
+    "-i",
+    required=True,
+    type=str,
+    help="Data CSV file that is used for training/inference; "
+    "can also take comma-separated training-validation pre-split CSVs;"
+    "can also take a path to folder with `data.csv`",
+)
+@click.option(
+    "--train/--infer",
+    "-t/--infer",
+    required=True,  # TODO: what if we make infer as default behavior if no param is passed?
+    help="If we run training or inference; for inference, "
+    "there needs to be a compatible model saved in '-model-dir'",
+)
+@click.option(
+    "--model-dir",
+    "-m",
+    type=click.Path(file_okay=False, dir_okay=True),
+    help="Training: Output directory to save intermediate files and model weights; "
+    "inference: location of previous training session output",
+)
+@click.option(
+    "--device",
+    "-d",
+    # TODO: Not sure it's worth to restrict this list. What about other devices?
+    #  GaNDLF guarantees to work properly with these two options, but
+    #  other values may be partially working also.
+    #  * GaNDLF code convert `-1` to `cpu` (i.e. it is expected somebody may pass -1)
+    #  * `cuda:0` should work also, isn't it? Just would not be treated as `cuda`
+    #  * Would `mps` work?
+    #  * int values (like `1`) - are they supported? (legacy mode for cuda https://pytorch.org/docs/stable/tensor_attributes.html#torch-device)
+    type=click.Choice(["cuda", "cpu"]),
+    required=True,  # FIXME: either keep default value, or set required flag
```

Comment:
Maybe defaulting here to one of the values depending on availability is worth considering?

Reply:
Hmm, I am not sure if putting a default here makes a lot of sense. If someone is running inference on optimized models, they might prefer `cpu`, but if they are running training, they might prefer `cuda`. No?

Reply:
Hmm, that's true, actually leaving it up to choice is not that big of a deal, at least user has to think about the device specification